builtin/providers/docker/resource_docker_container_funcs.go

```diff
@@ -0,0 +1,286 @@
+		Name: d.Get("name").(string),
+		Config: &dc.Config{
+			Image:      image,
+			Hostname:   d.Get("hostname").(string),
+			Domainname: d.Get("domainname").(string),
+		},
+	}
+
+	if v, ok := d.GetOk("env"); ok {
+		createOpts.Config.Env = stringSetToStringSlice(v.(*schema.Set))
+	}
+
+	if v, ok := d.GetOk("command"); ok {
+		createOpts.Config.Cmd = stringListToStringSlice(v.([]interface{}))
+	}
+
+	exposedPorts := map[dc.Port]struct{}{}
+	portBindings := map[dc.Port][]dc.PortBinding{}
+
+	if v, ok := d.GetOk("ports"); ok {
+		exposedPorts, portBindings = portSetToDockerPorts(v.(*schema.Set))
+	}
+	if len(exposedPorts) != 0 {
+		createOpts.Config.ExposedPorts = exposedPorts
+	}
+
+	volumes := map[string]struct{}{}
+	binds := []string{}
+	volumesFrom := []string{}
+
+	if v, ok := d.GetOk("volumes"); ok {
+		volumes, binds, volumesFrom, err = volumeSetToDockerVolumes(v.(*schema.Set))
+		if err != nil {
+			return fmt.Errorf("Unable to parse volumes: %s", err)
+		}
+	}
+	if len(volumes) != 0 {
+		createOpts.Config.Volumes = volumes
+	}
+
+	var retContainer *dc.Container
+	if retContainer, err = client.CreateContainer(createOpts); err != nil {
+		return fmt.Errorf("Unable to create container: %s", err)
+	}
+	if retContainer == nil {
+		return fmt.Errorf("Returned container is nil")
+	}
+
+	d.SetId(retContainer.ID)
+
+	hostConfig := &dc.HostConfig{
+		PublishAllPorts: d.Get("publish_all_ports").(bool),
+	}
+
+	if len(portBindings) != 0 {
+		hostConfig.PortBindings = portBindings
+	}
+
+	if len(binds) != 0 {
+		hostConfig.Binds = binds
+	}
+	if len(volumesFrom) != 0 {
+		hostConfig.VolumesFrom = volumesFrom
+	}
+
+	if v, ok := d.GetOk("dns"); ok {
+		hostConfig.DNS = stringSetToStringSlice(v.(*schema.Set))
+	}
+
+	if err := client.StartContainer(retContainer.ID, hostConfig); err != nil {
+		return fmt.Errorf("Unable to start container: %s", err)
+	}
+
+	return resourceDockerContainerRead(d, meta)
+}
+
+func resourceDockerContainerRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	client, err := config.NewClient()
+	if err != nil {
+		return fmt.Errorf("Unable to connect to Docker: %s", err)
+	}
+
+	apiContainer, err := fetchDockerContainer(d.Get("name").(string), client)
+	if err != nil {
+		return err
+	}
+
+	if apiContainer == nil {
+		// This container doesn't exist anymore
+		d.SetId("")
+
+		return nil
+	}
+
+	container, err := client.InspectContainer(apiContainer.ID)
+	if err != nil {
+		return fmt.Errorf("Error inspecting container %s: %s", apiContainer.ID, err)
+	}
+
+	if d.Get("must_run").(bool) && !container.State.Running {
+		return resourceDockerContainerDelete(d, meta)
+	}
+
+	return nil
+}
+
+func resourceDockerContainerUpdate(d *schema.ResourceData, meta interface{}) error {
+	if err := resourceDockerContainerDelete(d, meta); err != nil {
+		return fmt.Errorf("Error deleting container before recreation: %s", err)
+	}
+
+	return resourceDockerContainerCreate(d, meta)
+}
```

Comment:
If you can only nuke-and-recreate in your Update, you may just want to omit it and set all your configurable attributes to `ForceNew` - that way your plans will be more clear about when they're going to `-/+`